packages/core/lib/__tests__/migrate.spec.tsx

```diff
@@ -77,6 +77,158 @@ describe("migrate method", () => {
     expect(migrate(input, config)).toEqual(output);
   });
 
+  it("should migrate dynamic arrays of zones to slots when a user provides a migration function", () => {
+    const input: Data = {
```

Comment:
Doesn't add value here, but FWIW you can also type the data here by passing in props (i.e. `Data<Props>`)